### PR TITLE
State handling tweaks

### PR DESCRIPTION
**Pull request by neilalexander**
This tweaks how rejected events are handled in room state and also to not apply checks we can't complete to outliers.

### CODE REVIEW

File: roomserver/storage/interface.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ type Database interface {
 	// Look up the state entries for a list of string event IDs
 	// Returns an error if the there is an error talking to the database
 	// Returns a types.MissingEventError if the event IDs aren't in the database.
```
Comment (kegsay) [inline roomserver/storage/interface.go:81]: Need to comment how `excludeRejected` interacts with this. If all `eventIDs` are excluded, does it return `[]` or a `types.MissingEventError`?

File: roomserver/storage/sqlite3/events_table.go [hunk 2]
```diff
@@ -221,17 +237,18 @@ func (s *eventStatements) BulkSelectStateEventByID(
...
 		if err = rows.Scan(
 			&result.EventTypeNID,
 			&result.EventStateKeyNID,
 			&result.EventNID,
 		); err != nil {
 			return nil, err
 		}
+		results = append(results, result)
 	}
 	if i != len(eventIDs) {
```
Comment (kegsay) [inline roomserver/storage/sqlite3/events_table.go:253]: Missing `excludeRejected` check.

File: roomserver/internal/helpers/auth.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ func CheckForSoftFail(
 	var authStateEntries []types.StateEntry
 	var err error
 	if rewritesState {
-		authStateEntries, err = db.StateEntriesForEventIDs(ctx, stateEventIDs)
+		authStateEntries, err = db.StateEntriesForEventIDs(ctx, stateEventIDs, false)
```
Comment (kegsay) [inline roomserver/internal/helpers/auth.go:42]: Why do we include rejected events here? Seems dangerous?

File: roomserver/internal/helpers/auth.go [hunk 4]
```diff
@@ -97,7 +97,7 @@ func CheckAuthEvents(
 	authEventIDs []string,
 ) ([]types.EventNID, error) {
 	// Grab the numeric IDs for the supplied auth state events from the database.
-	authStateEntries, err := db.StateEntriesForEventIDs(ctx, authEventIDs)
+	authStateEntries, err := db.StateEntriesForEventIDs(ctx, authEventIDs, false)
```
Comment (kegsay) [inline roomserver/internal/helpers/auth.go:100]: Why do we include rejected events here? Seems dangerous?

**Review by kegsay — CHANGES_REQUESTED** [review]

File: roomserver/storage/interface.go (hunk 1 above)
Comment (neilalexander) [reply roomserver/storage/interface.go:81]: Updated the comment, it returns `[]`.

**Pull request merged**